src/main/java/org/tugraz/sysds/runtime/instructions/CPInstructionParser.java

```diff
@@ -362,7 +370,13 @@ public static CPInstruction parseSingleInstruction ( CPType cptype, String str )
 				
 			case Rand:
 				return DataGenCPInstruction.parseInstruction(str);
-				
+
+			case TensorGen:
+				return TensorGenCPInstruction.parseInstruction(str);
+
+			case TensorAggregateUnary:
```

Comment:
given the lack of an opcode, you can check on AggregateUnary the input data type and then either call the respective matrix or tensor operations.